extensions/cache/deployment/src/main/java/io/quarkus/cache/deployment/CacheNameResolver.java

```diff
@@ -0,0 +1,28 @@
+package io.quarkus.cache.deployment;
+
+import static io.quarkus.cache.deployment.CacheDeploymentConstants.CACHE_NAME_PARAMETER_NAME;
+import static io.quarkus.cache.deployment.CacheDeploymentConstants.GENERATED_CACHE_NAME_PREFIX;
+
+import java.util.stream.Collectors;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationValue;
+import org.jboss.jandex.MethodInfo;
+
+/**
+ * @author Gwenneg Lepage
+ */
+public class CacheNameResolver {
+
+    public static String resolve(AnnotationInstance cacheAnnotation) {
+        AnnotationValue explicitCacheName = cacheAnnotation.value(CACHE_NAME_PARAMETER_NAME);
+        return explicitCacheName == null ? generateCacheName(cacheAnnotation) : explicitCacheName.asString();
+    }
+
+    private static String generateCacheName(AnnotationInstance cacheAnnotation) {
+        MethodInfo method = cacheAnnotation.target().asMethod();
+        String signature = method.declaringClass().name() + "." + method.name() + "("
+                + method.parameters().stream().map(type -> type.name().toString()).collect(Collectors.joining(",")) + ")";
```

Comment:
This comes from [JCache javadoc](https://static.javadoc.io/javax.cache/cache-api/1.0.0/javax/cache/annotation/CacheResult.html#cacheName()).